packages/core/src/routes/profile/index.ts

```diff
@@ -2,32 +2,45 @@ import { z } from 'zod';
 
 import koaGuard from '#src/middleware/koa-guard.js';
 
+import { encryptUserPassword } from '../../libraries/user.utils.js';
+import { buildUserVerificationRecordById } from '../../libraries/verification.js';
 import koaOidcAuth from '../../middleware/koa-auth/koa-oidc-auth.js';
+import assertThat from '../../utils/assert-that.js';
 import type { ProfileRouter, RouterInitArgs } from '../types.js';
 
-/**
- * Authn stands for authentication.
- * This router will have a route `/authn` to authenticate tokens with a general manner.
- */
 export default function profileRoutes<T extends ProfileRouter>(
-  ...[router, { provider }]: RouterInitArgs<T>
+  ...[router, { provider, queries, libraries }]: RouterInitArgs<T>
 ) {
+  const {
+    users: { updateUserById },
+  } = queries;
+
   router.use(koaOidcAuth(provider));
 
-  // TODO: test route only, will implement a better one later
-  router.get(
-    '/profile',
+  router.post(
```

Comment:
Don't forget to trigger the user.updated webhooks.

Reply:
Added a TODO issue.